global_buffer/global_buffer_bs_generator.py

```diff
@@ -0,0 +1,214 @@
+from typing import List
+from abc import ABC, abstractmethod
+from gemstone.common.collections import DotDict
+from global_buffer.global_buffer_magma import GlobalBuffer
+import math
+import json
+
+
+def AXIConfigMap(config_addr_width, config_data_width,
+                 core_addr_width, feature_addr_width, reg_addr_width,
+                 byte_addressable=True):
+
+    _AXIConfigMap = DotDict()
+    _AXIConfigMap.config_addr_width = config_addr_width
+    _AXIConfigMap.config_data_width = config_data_width
+    _AXIConfigMap.core_addr_width = core_addr_width
+    _AXIConfigMap.feature_addr_width = feature_addr_width
+    _AXIConfigMap.reg_addr_width = reg_addr_width
+    _AXIConfigMap.byte_addressable = byte_addressable
+
+    return _AXIConfigMap
+
+
+# TODO(kongty): For now, it has nothing to do with RTL generation.
+class Glb():
+    def __init__(self, global_buffer: GlobalBuffer):
+        # TODO(kongty): Hand-written address mapping should migrate to generator
+        self.config_map = AXIConfigMap(global_buffer.axi_addr_width,
+                                       global_buffer.cfg_data_width,
+                                       2, 4, 4, True)
+        self.num_io = global_buffer.num_io
+        self.num_cfg = global_buffer.num_cfg
+        self.num_banks = global_buffer.num_banks
+
+        # TODO(kongty): Hand-written controllers should migrate to generator
+        self.io_ctrl = IOController(self, 1, self.num_io)
+        self.cfg_ctrl = ParCfgController(self, 2, self.num_cfg)
+        self.cores: List["GlbCore"] = [self.io_ctrl, self.cfg_ctrl]
+
+    def dump_all_config(self):
+        result = []
+        for core in self.cores:
+            for feature in core.features:
+                for reg in feature.registers.values():
+                    # Only dump name, addr, and default value
+                    result.append(reg)
```

Comment:
Also `range`? I'm confused about the comment here.

Reply:
Deprecated comment.
Will delete